skaffold/src/main/kotlin/com/google/container/tools/skaffold/run/SkaffoldRunConfigurations.kt

```diff
@@ -106,18 +111,33 @@ abstract class AbstractSkaffoldRunConfiguration(
 
         XmlSerializer.serializeInto(this, element)
     }
+    override fun checkConfiguration(){
+        checkConfiguration(project)
 
-    @Throws(RuntimeConfigurationException::class)
-    override fun checkRunnerSettings(
-        runner: ProgramRunner<*>,
-        runnerSettings: RunnerSettings?,
-        configurationPerRunnerSettings: ConfigurationPerRunnerSettings?
-    ) {
-        throw RuntimeConfigurationException("Hitting function2222")
     }
 
 
-    override fun checkSettingsBeforeRun(){
-        throw RuntimeConfigurationException("Hitting function")
+    companion object {
+        var skaffoldPath: Path? = Paths.get("skaffold")
+        @JvmStatic
+        fun checkConfiguration(project:Project) {
+            val projectBaseDir: VirtualFile? = project.guessProjectDir()
+            if (skaffoldPath?.exists() == true && projectBaseDir != null ) {
+                val testSkaffoldCommand = skaffoldPath.toString() + " " + (SkaffoldExecutorSettings.ExecutionMode.SINGLE_RUN.modeFlag)
```

Comment:
don't forget to update this to either do a version check. or update the path checking strategy using the code snippet suggestion I gave in another comment

Reply:
one option to help with testing:

move the path check to `SkaffoldFileService`. This is an applicationService defined in `skaffold.xml`. You can swap these out in unit tests with a mock using the `@TestService` annotation. I can walk you through this.

Reply:
sorry - I think `SkaffoldExecutorService` is the better place for this

Reply:
+1 on relocating this as a normal member function to `SkaffoldExecutorService`